File: frontend/src/types/document.ts
export interface Document {
  id: string;
  kb_id: string;
  filename: string;
  file_type: string;
  file_size: number;
  status: 'pending' | 'processing' | 'ready' | 'failed';
  error_message?: string;
  chunk_count: number;
  created_at: string;
}

export interface DeleteDocumentResponse {
  success: boolean;
  doc_id: string;
  milvus_deleted: boolean;
  bm25_deleted: boolean;
  warnings: string[];
}

export interface DeleteKnowledgeBaseResponse {
  success: boolean;
  kb_id: string;
  documents_deleted: number;
  milvus_deleted: boolean;
  bm25_deleted: boolean;
  warnings: string[];
}

export interface RebuildIndexResponse {
  status: 'completed' | 'partial' | 'failed';
  kb_id: string;
  document_count: number;
  success_documents: number;
  failed_documents: { doc_id: string; filename: string; error: string }[];
  chunk_count: number;
  bm25_chunk_count: number;
  warnings: string[];
}

export interface IndexStatusResponse {
  kb_id: string;
  document_count: number;
  chunk_count: number;
  bm25_chunk_count: number;
  bm25_index_exists: boolean;
  documents_by_status: Record<string, number>;
}

File: backend/app/models/document.py
from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: str
    kb_id: str
    filename: str
    file_type: str
    file_size: int
    status: str
    error_message: str | None = None
    chunk_count: int = 0
    created_at: str


class DocumentStatusResponse(BaseModel):
    status: str
    chunk_count: int
    error_message: str | None = None


class DeleteDocumentResponse(BaseModel):
    success: bool
    doc_id: str
    milvus_deleted: bool
    bm25_deleted: bool
    warnings: list[str] = []


class DeleteKnowledgeBaseResponse(BaseModel):
    success: bool
    kb_id: str
    documents_deleted: int
    milvus_deleted: bool
    bm25_deleted: bool
    warnings: list[str] = []


class RebuildIndexResponse(BaseModel):
    status: str  # "completed", "partial", "failed"
    kb_id: str
    document_count: int
    success_documents: int
    failed_documents: list[dict] = []
    chunk_count: int
    bm25_chunk_count: int
    warnings: list[str] = []


class IndexStatusResponse(BaseModel):
    kb_id: str
    document_count: int
    chunk_count: int
    bm25_chunk_count: int
    bm25_index_exists: bool
    documents_by_status: dict[str, int] = {}
